eth_account/account.py

```diff
@@ -229,6 +241,97 @@ def from_key(self, private_key):
+        :return: object with methods for signing and encrypting
+        :rtype: LocalAccount
+
+        .. code-block:: python
+
+            >>> from eth_account import Account
+            >>> Account.enable_unaudited_hdwallet_features()
+            >>> acct = Account.from_mnemonic(
+              "coral allow abandon recipe top tray caught video climb similar prepare bracket "
+              "antenna rubber announce gauge volume hub hood burden skill immense add acid")
+            >>> acct.address
+            '0x9AdA5dAD14d925f4df1378409731a9B71Bc8569d'
+
+            # These methods are also available: sign_message(), sign_transaction(), encrypt()
+            # They correspond to the same-named methods in Account.*
+            # but without the private key argument
+        """
+        if not self._use_unaudited_hdwallet_features:
+            raise AttributeError(
+                "The use of the Mnemonic features of Account is disabled by default until "
+                "its API stabilizes. To use these features, please enable them by running "
+                "`Account.enable_unaudited_hdwallet_features()` and try again."
+            )
+        seed = seed_from_mnemonic(mnemonic, passphrase)
+        if isinstance(account_index, int):
+            private_key = derive_ethereum_key(seed, account_index)
+            key = self._parsePrivateKey(private_key)
+            return LocalAccount(key, self)
+        if isinstance(account_index, list):
+            accounts = []
+            for idx in account_index:
+                private_key = derive_ethereum_key(seed, idx)
+                key = self._parsePrivateKey(private_key)
+                accounts.append(LocalAccount(key, self))
+            return accounts
+        raise ValidationError(
+            f"Account index must be either int or list of int, instead: '{account_index}'."
+        )
+
+    @combomethod
+    def create_with_mnemonic(self,
+                             passphrase="",
+                             num_words=12,
+                             language="english",
+                             account_index=0):
+        r"""
+        Creates a new private key, and returns it as a :class:`~eth_account.local.LocalAccount`,
+        alongside the mnemonic that can used to regenerate it using any BIP39-compatible wallet.
+
+        :param str passphrase: Extra passphrase to encrypt the seed phrase
+        :param int num_words: Number of words to use with seed phrase. Default is 12 words.
+                              Must be one of [12, 15, 18, 21, 24].
+        :param str language: Language to use for BIP39 mnemonic seed phrase.
+        :param account_index: Specify an alternate BIP44 account index used for deriving an
+            account from the seed using BIP32 HD wallet key derivation. Default is 0.
+        :type account_index: Either a positive integer, or a list of positive integers.
```

Comment:
Not quite, no.

I think I should refactor this so you can put in your own full path instead. "Index" refers to account index, so you would basically be making one account for `m/44'/60'/0'/0/45` and another for `m/44'/60'/0'/0/82`, assuming the same master seed.

I also wanted to explore adding a Signer that would be for mnemonics, so the master key was stored in the encrypted keyfile (and you can derive accounts from that). But then that got into making the encryption ABI into a generic base class so it could be reused.

There's a lot of options here.

Reply:
Oh, got it. That makes sense. Yeah, I think putting in your own path would make things a little more intuitive.